src/abstractdataframe/abstractdataframe.jl

```diff
@@ -518,27 +526,28 @@ function Base.convert{T}(::Type{Matrix{T}}, df::AbstractDataFrame)
     n, p = size(df)
     res = Array(T, n, p)
     idx = 1
-    for col in columns(df)
-        anyna(col) && error("DataFrame contains NAs")
-        copy!(res, idx, data(col))
+    for (name, col) in zip(names(df), columns(df))
+        anynull(col) && error("cannot convert a DataFrame containing null values to array (found for column $name)")
```

Comment:
Yeah, touching many parts of the code allows spotting many different bugs Please file issues for any improvements you can think of so that we don't forget them.